### PR TITLE
metrics: add tidb_rm_pool_concurrency

### LINKED ISSUE #41114: add metrics for pool's concurrency
## Enhancement

We need to metrics the concurrency of pool from dashboard.

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #41114

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
```
curl http://127.0.0.1:10080/metrics |grep tidb_rm
# HELP tidb_rm_pool_concurrency How many concurrency in the pool
# TYPE tidb_rm_pool_concurrency gauge
tidb_rm_pool_concurrency{type="backfill"} 32
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Yisaer — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

File: metrics/resourcemanager.go [hunk 1]
```diff
@@ -24,4 +24,12 @@ var (
 		Name:      "ema_cpu_usage",
 		Help:      "exponential moving average of CPU usage",
 	})
+	// PoolConcurrencyCounter means how much concurrency in the pool
+	PoolConcurrencyCounter = prometheus.NewGaugeVec(
```
Comment (YangKeao) [inline metrics/resourcemanager.go:28]: Will you modify the grafana dashboard in another PR?

Comment (hawkingrei) [reply metrics/resourcemanager.go:28]: Yes, I will.

**Pull request merged**